### PR TITLE
Stabilize flaky `test_dropout` and fix `rate=1.0` across backends

### PR DESCRIPTION
**Pull request by goyaladitya05**
This PR addresses statistical flakiness in `RandomCorrectnessTest::test_dropout` and fixes a crash/incorrect behavior when `rate=1.0` is used in various backends.

## Changes
- Increased from `(3, 5)` to `(10, 10)` in `test_dropout` to ensure statistical stability ($P(\text{all-dropped}) < 10^{-30}$).
- Added checks for `rate=0`, `rate=0.5` (with precision check), and `rate=1.0`.
- Added explicit `if rate == 1.0: return zeros` and `if rate == 0.0: return inputs` guards in TensorFlow, JAX, Torch, and NumPy backends.
    - **TensorFlow**: Fixed `ValueError` crash for `rate=1.0`.
    - **JAX/Torch/NumPy**: Prevented potential division-by-zero (`NaN`/`inf`) when calculating the scaling factor.

Keras `Dropout` layers allow `rate=1.0` in their validation logic, but backend implementations like TensorFlow's `stateless_dropout` enforce the range `[0, 1)`. By adding these guards at the Keras backend level, we ensure a consistent and robust API across all supported frameworks.

### Verification
Verified with `pytest keras/src/random/random_test.py -k test_dropout` across:
- **TensorFlow**: Passed
- **JAX**: Passed
- **Torch**: Passed
- **NumPy**: Passed
- **OpenVINO**: Passed 

Closes: #22311

### CODE REVIEW

File: keras/src/random/random_test.py [hunk 1]
```diff
@@ -105,11 +105,13 @@ def test_truncated_normal(self, seed, shape, mean, stddev):
         self.assertGreaterEqual(ops.max(res), mean - 2 * stddev)
 
     def test_dropout(self):
-        x = ops.ones((3, 5))
+        x = ops.ones((10, 10))
         self.assertAllClose(random.dropout(x, rate=0, seed=0), x)
-        x_res = random.dropout(x, rate=0.8, seed=0)
+        x_res = random.dropout(x, rate=0.5, seed=0)
         self.assertGreater(ops.max(x_res), ops.max(x))
```
[the remark this answers is not in the document]
Comment (goyaladitya05) [reply keras/src/random/random_test.py:111]: Good catch, added now.

**Comment by hertschuh** [comment]
> @hertschuh Could we please merge #22293 before any other PR releted to OpenVINO ? It has a fix in the file excluded_concrete_tests for the line endings, which would cause issues during rebase of other PRs. Please have a look once.

I noticed the issue. I think it's Windows linebreaks vs Unix. Can you send it as a separate PR? I'll merge it immediately. Thanks!

I do have questions about the seed change in #22293 which is why I don't want to merge it immediately.

**Comment by goyaladitya05** [comment]
> > @hertschuh Could we please merge #22293 before any other PR releted to OpenVINO ? It has a fix in the file excluded_concrete_tests for the line endings, which would cause issues during rebase of other PRs. Please have a look once.
> 
> I noticed the issue. I think it's Windows linebreaks vs Unix. Can you send it as a separate PR? I'll merge it immediately. Thanks!
> 
> I do have questions about the seed change in #22293 which is why I don't want to merge it immediately.

Fair, I'll send a PR soon. I just had a look at the files, this can be merged anyways since it does not touch that file. Thanks!
